### PR TITLE
Use SpeedSlider to control FrameRate in GameLab animation previews

### PR DESCRIPTION
**Pull request by caleybrock**
- Allow SpeedSlider to use a custom width
- Update JSDebugger and AnimationPreview to use appropriate widths and custom css
- Use SpeedSlider in animation preview to control frameRate
- Debounce events from SpeedSlider to update frameRate less often
- Change scale of frameRate to be between 0 and 20
  - Fix the 0 case so that there is no division by 0
- Remove mouseOver and MouseOut functionality when the animation is selected

framerate

todo in next PR: The frameRate set in the animation tab should save and effect the animations in the game space. (Currently it's not saved in Piskel either)

### CODE REVIEW

File: apps/src/gamelab/AnimationTab/AnimationListItem.jsx [hunk 1]
```diff
@@ -103,6 +114,20 @@ const AnimationListItem = React.createClass({
     this.props.setAnimationName(this.props.animationKey, event.target.value);
   },
 
+  convertFrameRateTo01(frameRate) {
+    return frameRate/20;
```
Comment (bcjordan) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: \+ spaces around `/`

File: apps/src/gamelab/AnimationTab/AnimationListItem.jsx [hunk 2]
```diff
@@ -103,6 +114,20 @@ const AnimationListItem = React.createClass({
     this.props.setAnimationName(this.props.animationKey, event.target.value);
   },
 
+  convertFrameRateTo01(frameRate) {
```
Comment (bcjordan) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: What does `01` mean in this context?

Comment (bcjordan) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: Ah, value between 0 to 1?

Comment (bcjordan) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: Apparently there's a lot of [debate](http://english.stackexchange.com/a/286524) and not a perfect word for it :D. lol @ `primantissa`. Think it's good as-is, few other options might be `0To1`, `Proportion` or `Fraction`.

File: apps/src/gamelab/AnimationTab/AnimationListItem.jsx [hunk 3]
```diff
@@ -103,6 +114,20 @@ const AnimationListItem = React.createClass({
...
+    return frameRate/20;
+  },
+
+  convert01ToFrameRate(value) {
+    return value * 20;
+  },
+
+  setAnimationFrameRate(frameRate) {
```
Comment (bcjordan) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: Should this parameter be named something `01`-y?

File: apps/src/templates/SpeedSlider.jsx [hunk 4]
```diff
@@ -27,9 +27,10 @@ const sliderImages = {
   lightRabbit: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABcAAAAPCAYAAAAPr1RWAAAABmJLR0QA/wD/AP+gvaeTAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH4AUCEjcsBQRNEAAAAX9JREFUOMuV0zFrVUEQBeBvwCJEEqukEeWhgiiCiIVYWATROqKFrUV6EfwH/gLbVBY2gpAilUQbCaKNRSQgGEnzCt8DwQRDurGZK5d4zbtv4LI7c3fP2Z09J/SMzHxY068R8bnPnhM9gc/X9ApWMhM+4DX8j6wXeETsZOZ8pXu4hZu4ivXM3IuInX/2mTLqFgMsFfgsXmHjKMHU4B1Ej3EZB3iJdxExmtiWzLyGO3hQpY9Yw27rlJtYwDesYJyZWxExmtTz+7jbym/Ut5+ZqxhjucAXas0SfmIUE6T3pGeH9mscYxVbEXEs+FvM9QSdwxs8L3V19zwzF3Gv0u84WfPZY0h+tGQ76tR5tWK5AA/qmms4izMlwSZ+13gO242hMnOxIYgq3K6Xbk75Hr9KAZu4hItFMo8vRXyhXDvAbtU3MIyIwwb8WeM2fMKwnCgiDjNzpkBPt1w6rHyA6+XaJtbxIlp6/gvYgHa8x8zR/+1akT3CKTyN9oIuwCkd2ybyB67+kmBPPxeyAAAAAElFTkSuQmCC"
 };
 
-const knobXMax = 135;
+let knobXMax = 135;
 const knobXMin = 15;
-const TRACK_LENGTH = knobXMax - knobXMin;
+const knobWidth = 16;
+let TRACK_LENGTH = knobXMax - knobXMin - 5;
```
Comment (bcjordan) [inline apps/src/templates/SpeedSlider.jsx]: s/`TRACK_LENGTH`/`trackLength` now that not a constant

Comment (bcjordan) [inline apps/src/templates/SpeedSlider.jsx]: (After reading below) another option could be to make this `DEFAULT_TRACK_LENGTH`, then in `componentDidMount` set `this.trackLength_ = DEFAULT_TRACK_LENGTH` and use `this.trackLength_` elsewhere.

**Comment by bcjordan** [comment]
LGTM! Nothing but nits. Awesome feature & nice implementation.

File: apps/src/gamelab/AnimationTab/AnimationListItem.jsx (hunk 2 above)
Comment (caleybrock) [inline apps/src/gamelab/AnimationTab/AnimationListItem.jsx]: yeah I had a hard time coming up with a good name - I think I'll go with fraction

**Comment by caleybrock** [comment]
^looked at the failures with Brian, going to merge.

**Pull request merged**